cve_bin_tool/output_engine.py

```diff
@@ -0,0 +1,39 @@
+import csv
+
+from datetime import datetime
+from builtins import hash
+
+
+class OutputEngine:
+    def __init__(self, filename=None, modules=None):
+        self.filename = filename
+        self.modules = modules
+
+        if self.filename == None:
+            now = datetime.now()
+            self.filename = str(hash(now)).strip("-")
+
+    def output_cves(self, outfile):
+
+        """ Output a list of CVEs
+        format is modules[checker_name][version] = dict{id: severity}
+        """
+        writer = csv.writer(outfile)
+        for modulename, versions in self.modules.items():
+            for version, cve_list in versions.items():
+                for cve_number, cve_severity in cve_list.items():
+                    row = [modulename, version, cve_number, cve_severity]
+                    writer.writerow(row)
+
+    def output_csv(self):
+
+        """ Generate a CSV file for list of CVE """
+
+        outfile = open(f"Output/{self.filename}.csv", "a")
+        writer = csv.writer(outfile)
+        writer.writerow(["Module Name", "Version", "CVE Number", "Severity"])
```

Comment:
Other than this line, the output in output_cves and output_csv is the same.  What about if we put this line into output_cves and have output_csv call output_cves with an argument?  I don't think anyone will mind having an explanatory line added to the console output here.